src/mumble/AudioOutput.cpp

```diff
@@ -588,29 +588,56 @@ bool AudioOutput::mix(void *outbuff, unsigned int frameCount) {
 						aop->pfVolume[s] = -1.0;
 				}
 
+				if (!aop->piOffset) {
+					aop->piOffset = std::make_unique<unsigned int []>(nchan);
+					for (unsigned int s = 0; s < nchan; ++s) {
+						aop->piOffset[s] = 0;
+					}
+				}
+
 				for (unsigned int s = 0; s < nchan; ++s) {
-					const float dot = bSpeakerPositional[s]
-										  ? connectionVec.x * speaker[s * 3 + 0] + connectionVec.y * speaker[s * 3 + 1]
-												+ connectionVec.z * speaker[s * 3 + 2]
-										  : 1.0f;
-					const float str   = svol[s] * calcGain(dot, len) * volumeAdjustment;
+					const float dot   = bSpeakerPositional[s]
+											? connectionVec.x * speaker[s * 3 + 0] + connectionVec.y * speaker[s * 3 + 1]
+                                                + connectionVec.z * speaker[s * 3 + 2]
+											: 1.0f;
+					// Volume on the ear opposite to the sound should never reach 0 in the real world.
+					// The gain is multiplied by 19/20 and 0.05 is added. This will have the effect
+					// of bringing the lowest value up to 0.05, while keeping the highest value at 1.
+					// E.g. calcGain() = 1; 1 * 19/20 = 0.95 + 0.05 = 1
+					// calcGain() = 1; 0 * 19/20 = 0 + 0.05 = 0.05
+					const float str   = svol[s] * (0.05 + 19 * calcGain(dot, len) / 20.0) * volumeAdjustment;
 					float *RESTRICT o = output + s;
 					const float old   = (aop->pfVolume[s] >= 0.0f) ? aop->pfVolume[s] : str;
 					const float inc   = (str - old) / static_cast< float >(frameCount);
 					aop->pfVolume[s]  = str;
+
+					// Calculates the ITD offset of the audio data this frame.
+					// Interaural Time Delay (ITD) is a small time delay between your ears
+					// depending on the sound source position on the horizonal plane and the
+					// distance between your ears.
+					//
+					// Offset for IAD is not applied directly, but rather the audio is squashed or
+					// stretched as needed to create a smooth transition between offsets.
+					// This prevents clicking / buzzing when the audio source or camera is moving.
```

Comment:
2 questions:
1. What is `IAD` (might also be worth to explain it in the same way you did for `ITD`)
2. The role of squashing/stretching the audio is not quite clear to me. Why is that needed again and where in the code does this happen? This terminology makes me think the audio data is somewhere multiplied somehow but that doesn't seem to be the case :thinking:

Reply:
`IAD` was a typo, it's supposed to say `ITD`.

The audio is squashed and stretched in order to transition between different offsets. What happens is the offset being added to the index (`currentOffset`) changes over time, resulting in the output audio being squashed or stretched so that at frame 0 it uses the offset of the previous chunk, and at the end of the output buffer it uses the output of the current chunk. This is to leave no "clicks" when the audio suddenly switches to a different offset value.

Maybe a better way to describe it would be that the offset value transitions smoothly across the entire chunk.

Reply:
Let my try to put that into my own words to see I understood it correctly:
Let's suppose that in one iteration the audio source has a certain position and based on that we calculate an offset in our audio buffer to be used for playback.
In the next iteration, the sound source has moved and therefore the target offset now is a different one. However our process is currently still tuned to the old offset so changing to the new offset abruptly causes clicking noises.
So instead of making one large jump between offsets we make many tiny jumps while filling the output buffer (thereby either duplicating or skipping certain samples).

Does that sound somewhat correct?

Reply:
Yep, that is correct.

Reply:
Alright then how about a description like this?
```
In order to smoothly transition between different offsets for ITD (that occur because the
offset is dependent on the audio source's position which can change), the offset is not
applied as a fixed value. Instead we incrementally switch from the old offset to the new one 
during processing of the current frame.
This avoid abrupt offset changes that cause audible audio artifacts like clicking/buzzing.
```

Reply:
> In order to smoothly transition between different offsets for ITD (required because the
offset is dependent on the audio source's position, which can change), the offset is not
applied as a fixed value. Instead, we gradually switch from the old offset to the new one 
when processing the current frame.
This avoids abrupt offset changes, which can cause audible audio artifacts like clicking/buzzing.